### PR TITLE
adding/updating examples for Virtual Machines (2.0 resources)

### LINKED ISSUE #1206: vm-joined-to-active-directory example doesn't work with version 1.4.0 of azurerm
Hi there,

I am trying to setup an active directory domain controller using this example and am unable to do so. The only difference that I've found is that this example references 

provider "azurerm" {
  version = "=1.1.2"
}

but i do not specify the version of the azurerm provider in my code.

### Terraform Version
v0.11.7

### Affected Resource(s)
Please list the resources as a list, for example:
- azurerm_virtual_machine

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
  os_profile_windows_config {
    provision_vm_agent        = true
    enable_automatic_upgrades = true

    additional_unattend_config {
      pass         = "oobeSystem"
      component    = "Microsoft-Windows-Shell-Setup"
      setting_name = "AutoLogon"
      content      = "<AutoLogon><Password><Value>${var.admin_password}</Value></Password><Enabled>true</Enabled><LogonCount>1</LogonCount><Username>${var.admin_username}</Username></AutoLogon>"
    }

    # Unattend config is to enable basic auth in WinRM, required for the provisioner stage.
    additional_unattend_config {
      pass         = "oobeSystem"
      component    = "Microsoft-Windows-Shell-Setup"
      setting_name = "FirstLogonCommands"
      content      = "${file("${path.module}/files/FirstLogonCommands.xml")}"
    }
  }

### Debug Output
Error: Error applying plan:

1 error(s) occurred:

* module.active-directory.azurerm_virtual_machine.primary-domain-controller: 1 error(s) occurred:

* azurerm_virtual_machine.primary-domain-controller: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The value of parameter windowsConfiguration.additionalUnattendContent.content is invalid." Target="windowsConfiguration.additionalUnattendContent.content"

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

### Expected Behavior
should have provisioned the azure vm and provided the additional unattend config.

### Actual Behavior
claims that the content is XML is invalid

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### PR DESCRIPTION
**Pull request by jackofallops**
also tidied up some HCL1 -> HCL2 in some places.
Fixes #1206 (for 2.0 resources)

### CODE REVIEW

File: examples/virtual-machines/linux/load-balanced/main.tf [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    source_address_prefix      = "*"
+    destination_port_range     = "443"
+    destination_address_prefix = azurerm_subnet.internal.address_prefix
+  }
+}
+
+resource "azurerm_lb" "example" {
+  name                = "TestLoadBalancer"
```
Comment (tombuildsstuff) [inline examples/virtual-machines/linux/load-balanced/main.tf]: could we use the prefix here:

```suggestion
  name                = "${var.prefix}-lb"
```

File: examples/virtual-machines/linux/load-balanced/main.tf [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  frontend_port                  = 443
+  backend_port                   = 443
+  frontend_ip_configuration_name = azurerm_lb.example.frontend_ip_configuration[0].name
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "example" {
+  count                   = local.instance_count
+  backend_address_pool_id = azurerm_lb_backend_address_pool.example.id
+  ip_configuration_name   = "primary"
+  network_interface_id    = azurerm_network_interface.main[count.index].id
```
Comment (tombuildsstuff) [inline examples/virtual-machines/linux/load-balanced/main.tf]: does this want to be:

```suggestion
  network_interface_id    = element(azurerm_network_interface.main.*.id, count.index)
```

File: examples/virtual-machines/linux/public-ip/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## Example: Basic Linux Virtual Machine
+
+This example provisions a basic Linux Virtual Machine with a public IP and a simple Network Security Group
```
Comment (tombuildsstuff) [inline examples/virtual-machines/linux/public-ip/README.md]:
```suggestion
This example provisions a basic Linux Virtual Machine with a Public IP and a Network Security Group
```

File: examples/virtual-machines/linux/load-balanced/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+## Example: Basic Linux Virtual Machine
+
+This example provisions a basic Linux Virtual Machine with a public IP and a simple Network Security Group
```
Comment (tombuildsstuff) [inline examples/virtual-machines/linux/load-balanced/README.md]: I think we can ditch the word "simple" here:

```suggestion
This example provisions a basic Linux Virtual Machine with a public IP and a Network Security Group
```

File: examples/virtual-machines/windows/vm-joined-to-active-directory/example.auto.tfvars [hunk 5]
```diff
@@ -0,0 +1,4 @@
+prefix         = "tf-ad-ex"
+location       = "westeurope"
+admin_username = "domadmin"
+admin_password = "Sup3r$ecRet_P@SsW0rd!"
```
Comment (tombuildsstuff) [inline examples/virtual-machines/windows/vm-joined-to-active-directory/example.auto.tfvars]: do we want to remove this file?

File: examples/virtual-machines/windows/vm-joined-to-active-directory/modules/domain-member/main.tf [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    }
+SETTINGS
+
+  protected_settings = <<SETTINGS
+    {
+        "Password": "${var.active_directory_password}"
+    }
+SETTINGS
+
+//  depends_on = [null_resource.wait-for-domain-to-provision]
```
Comment (tombuildsstuff) [inline examples/virtual-machines/windows/vm-joined-to-active-directory/modules/domain-member/main.tf]:
```suggestion
```

File: examples/virtual-machines/windows/vm-joined-to-active-directory/example.auto.tfvars (hunk 5 above)
Comment (jackofallops) [reply examples/virtual-machines/windows/vm-joined-to-active-directory/example.auto.tfvars]: I committed by mistake as I didn't realise *tfvars were not in the .gitignore, but then left it in place as an example of auto vars, happy to remove it though.

**Pull request merged**